### PR TITLE
README: reorganize and update readme of tikv

### PR DESCRIPTION
**Pull request by lilin90**
## What have you changed? (mandatory)

Reorganize README.md. The previous README is not logically clear and has some typos or redundant expressions.

## What are the type of the changes? (mandatory)

Improvement and typo fixes

## Does this PR affect documentation (docs/docs-cn) update? (mandatory)

No.

## Does this PR affect tidb-ansible update? (mandatory)

No.

### CODE REVIEW

**Comment by breeswish** [comment]
Seems that this PR also reverted some changes made previously: https://github.com/pingcap/tikv/pull/3009/files, i.e. the features section.

**Comment by lilin90** [comment]
@breeswish Yes, but I think the current display format is much clearer and more friendly to readers.

File: README.md [hunk 1]
```diff
@@ -1,54 +1,83 @@
...
+TiKV (The pronunciation is: /'taɪkeɪvi:/ tai-K-V, etymology: titanium) is a distributed Key-Value database powered by Rust and Raft. TiKV is based on the design of Google Spanner and HBase, but is much simpler without dependency on any distributed file system.
 
-TiKV has the following primary features:
+With the implementation of the Raft consensus algorithm in Rust and consensus state stored in RocksDB, TiKV guarantees data consistency. Placement Driver (PD), which is introduced to implement sharding, enables automatic data migration. The transaction model is similar to Google's Percolator with some performance improvements. TiKV also provides snapshot isolation (SI), snapshot isolation with lock (SQL: `SELECT ... FOR UPDATE`), and externally consistent reads and writes in distributed transactions.
```
Comment (Hoverbear) [inline README.md]: Maybe add a link to `pd`?

File: README.md [hunk 2]
```diff
@@ -1,54 +1,83 @@
...
 
-- **Geo-Replication:** TiKV uses [Raft](http://raft.github.io/) and the [Placement Driver](https://github.com/pingcap/pd/) to support Geo-Replication.
+## Key features
```
Comment (Hoverbear) [inline README.md]: I think this heading is unnecessary. It just repeats the line below.

File: README.md [hunk 3]
```diff
@@ -1,54 +1,83 @@
...
-When a node starts, the metadata of the Node, Store and Region are registered into PD. The status of each Region and Store is reported to PD regularly.
+When a node starts, the metadata of the Node, Store and Region are recorded into PD. The status of each Region and Store is reported to PD regularly.
 
+## Getting started
 
-## Your First Test Drive
+TiKV is a component of the [TiDB project](https://github.com/pingcap/tidb). To run TiKV you must build and run it with PD, which is used to manage the cluster. You can use TiKV together with TiDB. You can also use TiKV separately.
 
-We have a [Docker Compose](https://github.com/pingcap/tidb-docker-compose/) you can use to test out [TiKV](https://github.com/pingcap/tikv) and [TiDB](https://github.com/pingcap/tidb).
+We provide multiple deployment methods, but it is recommended to use the Ansible deployment for the production environment. The TiKV documentation is available on [TiKV's wiki page](https://github.com/pingcap/tikv/wiki/TiKV-Documentation).
```
Comment (Hoverbear) [inline README.md]: I'm not sure pointing people to production deployment guides is appropriate for a "Getting Started" section.

**Comment by Hoverbear** [comment]
Thanks for refining the README more! :tada:

**Comment by lilin90** [comment]
@Hoverbear PTAL again. Thanks!

**Pull request merged**